### PR TITLE
Update Tera example, macro compile_templates! is deprecated

### LINKED ISSUE #464: Update the tera example
The tera example uses the `compile_templates!` macro that has been removed in version 1.0 ([Changelog](https://github.com/Keats/tera/blob/master/CHANGELOG.md#100-2019-12-07)). It needs to be rewritten to compile with newer versions of tera.

Related: #460

### PR DESCRIPTION
**Pull request by edjroz**
Update Tera example to comply with Tera 1.5

Close #464

### CODE REVIEW

File: examples/templating/tera/src/main.rs [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
-    pub static ref TERA: Tera =
-        compile_templates!(concat!(env!("CARGO_MANIFEST_DIR"), "/templates/**/*"));
+    pub static ref TERA: Tera = {
+        let tera = match Tera::new(concat!(env!("CARGO_MANIFEST_DIR"), "/TEMPLATES/**/*")) {
+            Ok(t) => t,
+            Err(e) => {
+                println!("Parsing error(s): {}", e);
+                ::std::process::exit(1);
+            }
+        };
```
Comment (msrd0) [inline examples/templating/tera/src/main.rs:16]: Can't we just use `.expect` instead of the `match` block here?

**Review by msrd0 — CHANGES_REQUESTED** [review]
Hi, thanks for the PR! The travis test failed, can you fix that please?

File: examples/templating/tera/src/main.rs (hunk 1 above)
Comment (edjroz) [reply examples/templating/tera/src/main.rs:16]: We could but Ideally wouldn't we want to keep in on par with Tera's documentation?

Comment (msrd0) [reply examples/templating/tera/src/main.rs:16]: I think improving readability is more important than using code from Tera's documentation

File: examples/templating/tera/src/main.rs [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
 
 use gotham::state::State;
 use tera::{Context, Tera};
 
-// Assuming the Rust file is at the same level as the templates folder
-// we can get a Tera instance that way:
 lazy_static! {
-    pub static ref TERA: Tera =
-        compile_templates!(concat!(env!("CARGO_MANIFEST_DIR"), "/templates/**/*"));
+    pub static ref TERA: Tera = {
```
Comment (msrd0) [inline examples/templating/tera/src/main.rs:9]: I believe this block can now be made into a single expression

**Comment by edjroz** [comment]
> Hi, thanks for the PR! The travis test failed, can you fix that please?

are you able to replicate this error locally ? I'm running on Darwin and can't seem to replicate it by running `cargo test -j2 --all`

**Comment by msrd0** [comment]
Yes, I can replicate this issue on ArchLinux. I noticed you changed the template folder name, maybe you should double-check if that was actually intended?

**Review by msrd0 — APPROVED** [review]

**Pull request merged**